### PR TITLE
MAX17043 sensor integration

### PR DESCRIPTION
**Pull request by Vincent1964**
## Description:

Clean restart of my first PR for the MAX17043 power gauge.

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.9
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by sfromis** [comment]
Linky: https://github.com/arendst/Tasmota/pull/18739

**Comment by Vincent1964** [comment]
Tested on esp8266

**Comment by Vincent1964** [comment]
Should I increase xsns from 109 to 110 and update i2c devices?

**Comment by arendst** [comment]
Yes pls.

**Comment by Vincent1964** [comment]
> Yes pls.

Done!

**Comment by Vincent1964** [comment]
My PR is not going to next stage, did I miss something?

**Comment by arendst** [comment]
No. I need time.

**Comment by Vincent1964** [comment]
Okay Thanks for your reply!

File: tasmota/tasmota_xsns_sensor/xsns_110_max17043.ino [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+\*********************************************************************************************/
+
+bool Xsns109(uint32_t function) {
+if (!I2cEnabled(MAX17043_ADDRESS)) { return false; } 
+
+  switch (function) {
+    case FUNC_INIT:
+      Max17043Init(); 
+      break;
+    case FUNC_EVERY_SECOND:
```
Comment (arendst) [inline tasmota/tasmota_xsns_sensor/xsns_110_max17043.ino:114]: This construct needs changing! As it stands even with no sensr connected it keeps executing the following cases. You'll need to change the FUNC_INIT case to something disabling the below cases if no sensor found. Pls look at other sensor drivers for examples.

Comment (barbudor) [reply tasmota/tasmota_xsns_sensor/xsns_110_max17043.ino:114]: A typical method is to have a global pointer to all needed driver data that is initialized as nullptr
If you detect the device at INIT then you can 'calloc()' for your data and assign to that pointer
This provides 2 things
- do not use any fixed allocated RAM if the device is not detected
- a simple value to test against for all calls but INIT

Comment (Vincent1964) [reply tasmota/tasmota_xsns_sensor/xsns_110_max17043.ino:115]: Should be solved now by [2077d08](https://github.com/arendst/Tasmota/pull/18788/commits/2077d08ef75ec3e2138bb5a413e4d11ba0a6e111)

**Pull request merged**